multiversx_sdk_cli/contracts.py

```diff
@@ -192,6 +198,40 @@ def prepare_upgrade_transaction(self,
 
         return tx
 
+    def query_contract(self,
+                       contract_address: IAddress,
+                       proxy: INetworkProvider,
+                       function: str,
+                       arguments: List[Any],
+                       should_prepare_args: bool) -> List[Any]:
+        args = arguments if arguments else []
+        if should_prepare_args:
+            args = self._prepare_args_for_factory(args)
+
+        query_runner = QueryRunnerAdapter(proxy)
+        sc_query_controller = SmartContractQueriesController(query_runner, self._abi)
+
+        query = sc_query_controller.create_query(
+            contract=contract_address.to_bech32(),
+            function=function,
+            arguments=args
+        )
+
+        response = sc_query_controller.run_query(query)
+
+        if self._abi:
```

Comment:
This if-else is not consistent.
a)
With ABI you return the data-parts
Without ABI your return other stuff + data-parts

b) errors for bad queries
With ABI you `raise SmartContractQueryError`
Without ABI you return the Dict with error message

1/ What is the reason you return different types?
You can just return `sc_query_controller.parse_query_response(response)` but for NO abi case return parts as hex.
2/ Please add some unit tests for bad queries (wrong --function or whatever)
3/ I can't find test for contract query with abi

Reply:
Now, the `if-else` is gone. The querying is done in a `try-except` block so all the errors thrown are of type 
 `QueryContractError`. Did a small "fix" for the json encoder to convert all bytes to hex. Also added tests.

Reply:
Looks good to me